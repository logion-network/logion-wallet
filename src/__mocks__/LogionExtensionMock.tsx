export let _enabledApp: string | null = null;

export function getAccounts(appName: string, _extensions?: string[]): Promise<any[]> {
    _enabledApp = appName;
    return Promise.resolve(_accounts);
}

export function enabledApp(): string | null {
    return _enabledApp;
}

export let isWeb3Injected = true;

export function setWeb3Injected(value: boolean) {
    isWeb3Injected = value;
}

let _accounts: any[] = [];

export function updateInjectedAccounts(accounts: any[]) {
    _accounts = accounts;
}

const extensions: Record<string, any> = {
    lockedSigner: {
        signer: "lockedSigner"
    }
};

export function web3FromAddress(signerId: string): Promise<any> {
    return Promise.resolve(extensions[signerId]);
}

export function teardown() {
    _enabledApp = null;
    _accounts = [];
}

export function setSigner(signerId: string, signer: any) {
    extensions[signerId] = {
        signer
    };
}

export class ExtensionSigner {

}

export function isExtensionAvailable() {
    return true;
}

let metamaskEnabled = false;

export function enableMetaMask(appName: string) {
    return Promise.resolve(metamaskEnabled)
}

export function setMetamaskEnabled(value: boolean) {
    metamaskEnabled = value;
}

export function allMetamaskAccounts() {
    if(metamaskEnabled) {
        return Promise.resolve([{
            address: "0xa6db31d1aee06a3ad7e4e56de3775e80d2f5ea84",
        }]);
    } else {
        return Promise.resolve([]);
    }
}
