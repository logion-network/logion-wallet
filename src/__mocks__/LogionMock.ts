import { AccountType, ValidAccountId, LogionNodeApiClass, LegalOfficerCase, Hash } from "@logion/node-api";
import { H256 } from "@logion/node-api/dist/types/interfaces";
import { It, Mock } from "moq.ts";
import type { Text, Compact, u128 } from '@polkadot/types-codec';
import type { SubmittableExtrinsic } from '@polkadot/api/promise/types';
import { TEST_WALLET_USER } from "src/wallet-user/TestData";

export const DEFAULT_LEGAL_OFFICER = mockValidPolkadotAccountId("5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"); // Alice
export const ANOTHER_LEGAL_OFFICER = mockValidPolkadotAccountId("5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"); // Bob
export const A_THIRD_LEGAL_OFFICER = mockValidPolkadotAccountId("5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"); // Charlie

export function mockValidPolkadotAccountId(address: string): ValidAccountId {
    return mockValidAccountId(address, "Polkadot");
}

export function mockValidAccountId(address: string, type: AccountType): ValidAccountId {
    return {
        address,
        type,
        toKey: () => `${type}:${address}`,
    } as ValidAccountId;
}

export const api = new Mock<LogionNodeApiClass>();

api.setup(instance => instance.queries.getValidAccountId).returns((address: string, _type: string) => mockValidPolkadotAccountId(address));

const localPeerId = new Mock<Text>();
localPeerId.setup(instance => instance.toString()).returns("Mock peer ID");
api.setup(instance => instance.polkadot.rpc.system.localPeerId()).returnsAsync(localPeerId.object());

api.setup(instance => instance.polkadot.createType(It.IsAny())).returnsAsync({});

const locId = new Mock<Compact<u128>>();
api.setup(instance => instance.adapters.toLocId(It.IsAny())).returns(locId.object());

const hash = new Mock<H256>();
api.setup(instance => instance.adapters.toH256(It.IsAny())).returns(hash.object());

export function setupApiMock(setupFunction: (api: Mock<LogionNodeApiClass>) => void) {
    setupFunction(api);
}

export function mockSubmittable(): Mock<SubmittableExtrinsic> {
    return new Mock<SubmittableExtrinsic>();
}

export const CLOSED_IDENTITY_LOC_ID = "85833363768713528858922097642089825569";

export const CLOSED_IDENTITY_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Identity',
    closed: true,
    files: [
        {
            hash: Hash.fromHex("0xf35e4bcbc1b0ce85af90914e04350cce472a2f01f00c0f7f8bc5c7ba04da2bf2"),
            nature: Hash.fromHex("0x636af5586476273b2b9d9fc318e0d1f124f4e9591b7218da4e4e27f6b071cc94"),
            submitter: TEST_WALLET_USER,
            size: 42n,
            acknowledged: true,
        }
    ],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}

export const OPEN_IDENTITY_LOC_ID = "195914524858768213081425411950368569411";

export const OPEN_IDENTITY_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Identity',
    closed: false,
    files: [],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}

export const CLOSED_COLLECTION_LOC_ID = "195914524858768213081425411950368569411";

export const CLOSED_COLLECTION_LOC: LegalOfficerCase = {
    owner: DEFAULT_LEGAL_OFFICER.address,
    requesterAddress: mockValidPolkadotAccountId("5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW"),
    locType: 'Collection',
    closed: true,
    files: [],
    metadata: [],
    links: [],
    collectionCanUpload: false,
}
