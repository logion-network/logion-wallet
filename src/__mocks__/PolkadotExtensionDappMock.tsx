export let _enabledApp: string | null = null;

export function enableExtensions(appName: string, callback: (accounts: any[]) => void): Promise<(consumer: (accounts: any[]) => void) => void> {
    _enabledApp = appName;
    _accountsCallback = callback;
    return Promise.resolve(web3AccountsSubscribe);
}

export function enabledApp(): string | null {
    return _enabledApp;
}

export let isWeb3Injected = true;

export function setWeb3Injected(value: boolean) {
    isWeb3Injected = value;
}

let _accountsCallback: ((accounts: any[]) => void) | null = null;

export function web3AccountsSubscribe(callback: (accounts: any[]) => void): Promise<void> {
    _accountsCallback = callback;
    return Promise.resolve();
}

export function accountsCallback() {
    return _accountsCallback;
}

export function updateInjectedAccounts(accounts: any[]) {
    _accountsCallback!(accounts);
}

const extensions: Record<string, any> = {
    lockedSigner: {
        signer: "lockedSigner"
    }
};

export function web3FromAddress(signerId: string): Promise<any> {
    return Promise.resolve(extensions[signerId]);
}

export function teardown() {
    _enabledApp = null;
    _accountsCallback = null;
}

export function setSigner(signerId: string, signer: any) {
    extensions[signerId] = {
        signer
    };
}

export class ExtensionSigner {

}

export function isExtensionAvailable() {
    return true;
}

export function enableMetaMask(appName: string) {
    return Promise.resolve(false)
}

export function allMetamaskAccounts() {
    return Promise.resolve([])
}
