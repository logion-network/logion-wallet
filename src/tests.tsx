import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };

export async function clickByName(name: string | RegExp | ((accessibleName: string, element: Element) => boolean)) {
    let button: HTMLElement;
    await waitFor(() => button = screen.getByRole("button", { name }));
    await userEvent.click(button!);
}

export async function typeByLabel(label: string, value: string) {
    let input: HTMLElement;
    await waitFor(() => input = screen.getByLabelText(label));
    await userEvent.type(input!, value);
}
