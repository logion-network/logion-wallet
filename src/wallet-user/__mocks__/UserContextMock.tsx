import { LogionNodeApiClass, Fees } from '@logion/node-api';

import { twoLegalOfficers } from "../../common/TestData";
import { TEST_WALLET_USER } from '../TestData';
import { AccountTokens, DirectoryClient, NoProtection, ProtectionState, LegalOfficerClass, DraftRequest } from '@logion/client';
import { AxiosFactory } from '@logion/client/dist/AxiosFactory.js';
import { ComponentFactory } from '@logion/client/dist/ComponentFactory.js';
import { LegalOfficerEndpoint, LogionClientConfig, SharedState } from '@logion/client/dist/SharedClient.js';
import { NetworkState } from '@logion/client/dist/NetworkState.js';
import { UserContext, ProtectionRequestParams } from '../UserContext';
import { BalanceState } from '@logion/client/dist/Balance.js';
import {
    LocsState,
    PendingRequest,
    ClosedLoc,
    RejectedRequest,
    ClosedCollectionLoc,
    OpenLoc, VoidedCollectionLoc, VoidedLoc
} from "@logion/client";
import { LocType, ValidAccountId } from "@logion/node-api";
import { Mock } from 'moq.ts';

export let createTokenRequest = () => null;

export let refreshRequests = jest.fn();

export let createProtectionRequest = jest.fn().mockReturnValue(Promise.resolve());

export function estimateFeesCreateProtectionRequest(_: ProtectionRequestParams): Promise<Fees> {
    return Promise.resolve(Fees.zero());
}

export let activateProtection = jest.fn().mockReturnValue(Promise.resolve());

export let claimRecovery = jest.fn().mockReturnValue(Promise.resolve());

export let api = new Mock<LogionNodeApiClass>();

export const DEFAULT_SHARED_STATE: SharedState = {
    axiosFactory: {} as AxiosFactory,
    componentFactory: {} as ComponentFactory,
    config: {

    } as LogionClientConfig,
    directoryClient: {} as DirectoryClient,
    legalOfficers: twoLegalOfficers,
    allLegalOfficers: twoLegalOfficers,
    networkState: {} as NetworkState<LegalOfficerEndpoint>,
    nodeApi: api.object(),
    tokens: {} as AccountTokens,
    currentAccount: {
        address: TEST_WALLET_USER,
        type: "Polkadot"
    } as unknown as ValidAccountId,
}

export let protectionState: ProtectionState | undefined = {} as NoProtection;

export function setProtectionState(state: ProtectionState | undefined) {
    protectionState = state;
}

export let recoveredBalanceState: BalanceState | undefined = {} as BalanceState;

export function setRecoveredBalanceState(state: BalanceState | undefined) {
    recoveredBalanceState = state;
}

export let mutateRecoveredBalanceState = jest.fn().mockReturnValue(Promise.resolve());

export let locsState: Partial<LocsState> | undefined = undefined;

export function setOpenedLocRequests(requests: any[]) {
    locsState = {
        get openLocs(): Record<LocType, OpenLoc[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setClosedLocRequests(requests: any[]) {
    locsState = {
        get closedLocs(): Record<LocType, (ClosedLoc | ClosedCollectionLoc)[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setVoidedLocs(requests: any[]) {
    locsState = {
        get voidedLocs(): Record<LocType, (VoidedLoc | VoidedCollectionLoc)[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setPendingLocRequests(requests: any[]) {
    locsState = {
        get pendingRequests(): Record<LocType, PendingRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setRejectedLocRequests(requests: any[]) {
    locsState = {
        get rejectedRequests(): Record<LocType, RejectedRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setDraftLocRequests(requests: any[]) {
    locsState = {
        get draftRequests(): Record<LocType, DraftRequest[]> {
            return {
                "Transaction": requests,
                "Collection": [],
                "Identity": [],
            }
        }
    }
}

export function setOpenVerifiedIssuerLocs(transactions: any[], collections: any[]) {
    locsState = {
        get openVerifiedIssuerLocs(): Record<LocType, OpenLoc[]> {
            return {
                "Transaction": transactions,
                "Collection": collections,
                "Identity": [],
            }
        }
    }
}
export function setClosedVerifiedIssuerLocs(transactions: any[], collections: any[]) {
    locsState = {
        get closedVerifiedIssuerLocs(): Record<LocType, ClosedLoc[]> {
            return {
                "Transaction": transactions,
                "Collection": collections,
                "Identity": [],
            }
        }
    }
}

export function setHasNonVoidIdentityLoc(legalOfficers: LegalOfficerClass[]) {
    locsState = {
        legalOfficersWithNonVoidIdentityLoc: legalOfficers
    }
}

export function setLocsState(locsStateMock: LocsState) {
    locsState = locsStateMock;
}

export let mutateLocsState = jest.fn().mockReturnValue(Promise.resolve());

export function useUserContext() {
    return {
        refreshRequests,
        createProtectionRequest,
        estimateFeesCreateProtectionRequest,
        activateProtection,
        claimRecovery,
        protectionState,
        recoveredBalanceState,
        mutateRecoveredBalanceState,
        locsState,
        mutateLocsState,
    } as unknown as Partial<UserContext>;
}

export function setCreateTokenRequest(callback: any) {
    createTokenRequest = callback;
}

export function setRefreshRequests(func: any) {
    refreshRequests = func;
}

export function setCreateProtectionRequest(callback: any) {
    createProtectionRequest = callback;
}

export function setMutateLocsState(callback: any) {
    mutateLocsState = callback;
}

export function UserContextProvider() {
    return (
        <div/>
    );
}
