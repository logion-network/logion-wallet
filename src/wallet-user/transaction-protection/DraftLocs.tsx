import { LocType } from "@logion/node-api/dist/Types";
import { useMemo } from "react";
import { useNavigate } from "react-router-dom";

import IdentityLocRequestDetails from "../../components/identity/IdentityLocRequestDetails";
import { useUserContext } from "../UserContext";
import Table, { Cell, EmptyTableMessage, DateTimeCell } from '../../common/Table';
import LocStatusCell from '../../common/LocStatusCell';
import LegalOfficerName from '../../common/LegalOfficerNameCell';
import Loader from '../../common/Loader';
import ButtonGroup from "src/common/ButtonGroup";
import Button from "src/common/Button";
import { locDetailsPath } from "../UserRouter";

export interface Props {
    locType: LocType
}

export default function DraftLocs(props: Props) {
    const { locsState } = useUserContext();
    const navigate = useNavigate();
    const { locType } = props;

    const data = useMemo(() => locsState?.draftRequests[locType].map(locState => locState.data()) || [], [ locsState, locType ]);

    if(locsState === null || locsState?.draftRequests === undefined) {
        return <Loader />;
    }

    return (
        <Table
            columns={[
                {
                    header: "Legal officer",
                    render: locData => <LegalOfficerName address={ locData.ownerAddress } />,
                    renderDetails: locType === 'Identity' ? locData => <IdentityLocRequestDetails personalInfo={ locData }/> : undefined,
                    align: 'left',
                },
                {
                    "header": "Description",
                    render: locData => <Cell content={ locData.description } overflowing tooltipId="description" />,
                    align: 'left',
                },
                {
                    header: "Status",
                    render: locData => <LocStatusCell status={ locData.status }/>,
                    width: "140px",
                },
                {
                    header: "Creation date",
                    render: locData => <DateTimeCell dateTime={ locData.createdOn || null } />,
                    width: '200px',
                    align: 'center',
                },
                {
                    header: "Action",
                    render: locData => (
                        <ButtonGroup>
                            <Button
                                onClick={ () => navigate(locDetailsPath(locData.id, locData.locType)) }
                            >
                                Edit
                            </Button>
                        </ButtonGroup>
                    ),
                }
            ]}
            data={ data }
            renderEmpty={ () => <EmptyTableMessage>No draft request</EmptyTableMessage> }
        />
    );
}
