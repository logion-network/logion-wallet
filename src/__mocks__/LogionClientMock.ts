import { DateTime } from "luxon";
import { ALAIN, GUILLAUME, PATRICK, threeLegalOfficers } from "../common/TestData";
import { TEST_WALLET_USER } from "../wallet-user/TestData";
export { toIsoString, fromIsoString } from "@logion/client/dist/DateTimeUtil.js";
import { api } from "./LogionMock";
import { VerifiedIssuerWithSelect } from "@logion/client/dist/Loc";
import { Hash } from "@logion/node-api";

export const axiosMock = {
    post: jest.fn().mockReturnValue(undefined),
};

export class LogionClient {

    static create() {
        return new LogionClient()
    }

    useTokens(tokens: any) {
        this.tokens = tokens;
        return this;
    }

    withCurrentAddress() {
        return this;
    }

    tokens = new AccountTokens();

    isLegalOfficer(address: string) {
        return address === PATRICK.address || address === GUILLAUME.address || address === ALAIN.address;
    }

    isRegisteredLegalOfficer(address: string) {
        return address === PATRICK.address || address === GUILLAUME.address || address === ALAIN.address;
    }

    legalOfficers = threeLegalOfficers;

    isValidAddress() {
        return true;
    }

    isProtected() {
        return true;
    }

    locsState() {
        return {

            findById: () => undefined
        };
    }

    authenticate() {
        return this;
    }

    buildAxios() {
        return axiosMock;
    }

    public = {
        fees: {
            estimateWithoutStorage: () => Promise.resolve({ inclusionFee: 42n, totalFee: 42n }),
            estimateAddFile: () => Promise.resolve({ inclusionFee: 42n, storageFee: 32n, totalFee: 74n }),
        },
        findCollectionLocItemById: () => { isAuthenticatedTokenOwner: () => true },
    };
    currentAddress = TEST_WALLET_USER;

    logionApi = api.object();

    getLegalOfficer(address: string) {
        if(address === PATRICK.address) {
            return threeLegalOfficers[0];
        } else if(address === GUILLAUME.address) {
            return threeLegalOfficers[1];
        } else if(address === ALAIN.address) {
            return threeLegalOfficers[2];
        } else {
            throw new Error();
        }
    }
}

export class AccountTokens {

    loadTokens() {
        return this;
    }

    cleanUp() {
        return this;
    }

    addresses = [ TEST_WALLET_USER ];

    get(account: any) {
        return ({
            value: `some-token-value-for-${account.address}`,
            expirationDateTime: DateTime.now().plus({hours: 1})
        });
    }
}

export class LocRequestState {
    data: any;
    locsState: any;
    async refresh(): Promise<LocRequestState> {
        return this;
    }
    getCurrentState() {
        return this;
    }
    isRequester(account: any) {
        return false;
    }
    isOwner(account: any) {
        return false;
    }
}

export class ClosedCollectionLoc extends LocRequestState {
    requestSof: jest.Mock<Promise<EditableRequest>> | undefined;
}

export class ClosedLoc extends LocRequestState {

    legalOfficer: {
        requestVote: any,
    } = {
        requestVote: jest.fn(),
    };
}

export class EditableRequest extends LocRequestState {
    addMetadata: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteMetadata: jest.Mock<Promise<EditableRequest>> | undefined;
    addFile: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteFile: jest.Mock<Promise<EditableRequest>> | undefined;
    addLink: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteLink: jest.Mock<Promise<EditableRequest>> | undefined;
}

export class DraftRequest extends EditableRequest {

}

export class PendingRequest extends LocRequestState {

    legalOfficer: {
        accept: any,
        acceptCollection: any,
    } = {
        accept: jest.fn(),
        acceptCollection: jest.fn(),
    };
}

export class OpenLoc extends EditableRequest {
    data: any;
    locsState: any;
    addMetadata: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteMetadata: jest.Mock<Promise<EditableRequest>> | undefined;
    addFile: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteFile: jest.Mock<Promise<EditableRequest>> | undefined;
    addLink: jest.Mock<Promise<EditableRequest>> | undefined;
    deleteLink: jest.Mock<Promise<EditableRequest>> | undefined;
    requestSof: jest.Mock<Promise<EditableRequest>> | undefined;
    legalOfficer: {
        close: any,
        voidLoc: any,
        getVerifiedIssuers: () => Promise<VerifiedIssuerWithSelect[]>,
        setCollectionFileRestrictedDelivery: (params: any) => Promise<OpenLoc>,
        canClose: (autoAck: boolean) => boolean,
        canAutoAck: () => boolean,
    } = {
        close: jest.fn(),
        voidLoc: jest.fn(),
        getVerifiedIssuers: () => Promise.resolve(issuers),
        setCollectionFileRestrictedDelivery: () => Promise.resolve(this),
        canClose: () => true,
        canAutoAck: () => true,
    };
}

export function setVerifiedIssuers(array: VerifiedIssuerWithSelect[]) {
    issuers = array;
}

let issuers: VerifiedIssuerWithSelect[] = [];

export class LocsState {
}

let expectedFileHash = Hash.of("");

export function setExpectedFileHash(hash: Hash) {
    expectedFileHash = hash;
}

export class HashOrContent {

    static async fromContentFinalized(file: File) {
        return new HashOrContent(file);
    }

    static fromContent(file: File) {
        return new HashOrContent(file);
    }

    constructor(file: File) {
        this.file = file;
    }

    private file: File;

    get contentHash(): Hash {
        return expectedFileHash;
    }

    get content(): File {
        return this.file;
    }

    finalize() {

    }
}

export class ReadOnlyLocState extends LocRequestState {

}
