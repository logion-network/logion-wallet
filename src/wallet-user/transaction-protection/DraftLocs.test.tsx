jest.mock('../../common/CommonContext');
jest.mock('../../common/Model');
jest.mock('../../wallet-user/UserContext');
jest.mock('../../logion-chain');

import { render } from '../../tests';
import DraftLocs from './DraftLocs';
import { setDraftLocRequests } from "../__mocks__/UserContextMock";

describe("DraftLocs", () => {

    it("renders null with no data", () => {
        const tree = render(<DraftLocs locType="Transaction" />);
        expect(tree).toMatchSnapshot();
    });

    it("renders empty list", () => {
        setDraftLocRequests([]);
        const tree = render(<DraftLocs locType="Transaction" />);
        expect(tree).toMatchSnapshot();
    });

    it("renders pending requests", () => {
        setDraftLocRequests([
            {
                data: () => {
                    return {
                        id: "6378b339-8f4a-486a-bf5e-6f34b951456a",
                        ownerAddress: "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
                        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
                        description: "LOC description",
                        status: "REVIEW_PENDING"
                    }
                }
            }
        ]);
        const tree = render(<DraftLocs locType="Transaction" />);
        expect(tree).toMatchSnapshot();
    });
});
