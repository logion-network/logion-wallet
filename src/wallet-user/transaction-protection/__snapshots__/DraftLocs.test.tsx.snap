// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftLocs renders empty list 1`] = `
<div
  className="Table constrained-row-height"
>
  <style>
    
            .Table .body .Row .Col.split-after::before {
                background-color: #152665;
            }
            
  </style>
  <div
    className="header"
    style={
      Object {
        "backgroundColor": "#0c163d",
        "color": "#ffffff",
      }
    }
  >
    <div
      className="Row"
    >
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Legal officer
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Description
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "140px",
          }
        }
      >
        Status
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Creation date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Action
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div
      className="Row empty-data"
      style={
        Object {
          "backgroundColor": "#152665",
          "color": "#ffffff",
        }
      }
    >
      <div
        className="empty-message"
      >
        No draft request
      </div>
    </div>
  </div>
</div>
`;

exports[`DraftLocs renders null with no data 1`] = `
<div
  className="spinner-border"
/>
`;

exports[`DraftLocs renders pending requests 1`] = `
<div
  className="Table constrained-row-height"
>
  <style>
    
            .Table .body .Row .Col.split-after::before {
                background-color: #152665;
            }
            
  </style>
  <div
    className="header"
    style={
      Object {
        "backgroundColor": "#0c163d",
        "color": "#ffffff",
      }
    }
  >
    <div
      className="Row"
    >
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Legal officer
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "left",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Description
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "140px",
          }
        }
      >
        Status
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "200px",
          }
        }
      >
        Creation date
      </div>
      <div
        className="Col"
        style={
          Object {
            "textAlign": "center",
            "width": "calc(( 100% - 340px ) / 3)",
          }
        }
      >
        Action
      </div>
    </div>
  </div>
  <div
    className="body"
  >
    <div>
      <div
        className="Row "
        style={
          Object {
            "color": "#ffffff",
          }
        }
      >
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 340px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell"
            style={Object {}}
          >
            Patrick Gielen
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "left",
              "width": "calc(( 100% - 340px ) / 3)",
            }
          }
        >
          <div
            className="regular-cell overflowing"
            style={Object {}}
          >
            <span
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
            >
              LOC description
            </span>
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "140px",
            }
          }
        >
          <span
            className="StatusCell"
            style={
              Object {
                "borderColor": "#ff9b3f",
              }
            }
          >
            <img
              alt="pending"
              height="20px"
              src="/assets/pending.svg"
            />
             
            <span
              style={
                Object {
                  "color": "#ff9b3f",
                  "textTransform": undefined,
                }
              }
            >
              Pending
            </span>
          </span>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "200px",
            }
          }
        >
          <div
            className="regular-cell"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            -
          </div>
        </div>
        <div
          className="Col"
          style={
            Object {
              "backgroundColor": "#152665",
              "fontSize": undefined,
              "textAlign": "center",
              "width": "calc(( 100% - 340px ) / 3)",
            }
          }
        >
          <div
            className="ButtonGroup"
          >
            <div
              className="btn-group"
              role="group"
            >
              <button
                className="Button btn btn-primary"
                disabled={false}
                onClick={[Function]}
                style={Object {}}
                type="button"
              >
                Edit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
